core/src/test/java/org/elasticsearch/index/IndexServiceTests.java

```diff
@@ -187,11 +189,18 @@ protected String getThreadPool() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
+        BiFunction<AtomicReference<CountDownLatch>, CountDownLatch, CountDownLatch> swapAndReturn = (ref, newLatch) -> {
+            CountDownLatch downLatch = ref.get();
+            ref.set(newLatch);
+            return downLatch;
+        };
         latch.get().await();
-        latch.set(new CountDownLatch(1));
+        swapAndReturn.apply(latch, new CountDownLatch(1));
```

Comment:
Can't we just keep the previous line here?  We don't use the returned (old latch) result?